### PR TITLE
Refactor disruption e2e test and add some eviction test cases.

### PR DESCRIPTION
**Pull request by mml**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32160)

### CODE REVIEW

**Comment by mml** [comment]
@davidopp this is not all test cases, but these are the ones I wrote last week.

File: test/e2e/disruption.go [hunk 1]
```diff
@@ -66,73 +68,98 @@ var _ = framework.KubeDescribe("DisruptionController", func() {
...
+			minAvailable: intstr.FromInt(2),
+			podCount:     1,
+			shouldDeny:   true,
+		}, {
+			description:  "enough pods, absolute",
+			minAvailable: intstr.FromInt(2),
+			podCount:     2,
+			shouldDeny:   false,
+		}, {
+			description:    "enough pods, replicaSet, percentage",
```
Comment (davidopp) [inline test/e2e/disruption.go]: Just for completeness, can you add one more test case?

```
"too few pods, replicaSet, percentage",
instr.FromString("100%")
replicaSetSize: 1
shouldDeny: true
```

**Comment by davidopp** [comment]
just one small comment

File: test/e2e/disruption.go (hunk 1 above)
Comment (mml) [inline test/e2e/disruption.go]: I had been trying to figure out how to force a controller to be at <100%.  The best I can come up with is to put an impossible node selector on it to make 0% of them be ready.  WDYT?

**Comment by mml** [comment]
@davidopp PTAL added two meaningful tests for the percentage case.. thanks for the hostPort suggestion.

**Comment by davidopp** [comment]
LGTM

**Comment by davidopp** [comment]
I'm not going to mark this as cherrypick since it's just a test and will go into HEAD.

**Comment by davidopp** [comment]
Oh, actually looks like the test itself is failing, with "resource name may not be empty"

Removing LGTM.

**Comment by mml** [comment]
Test is now passing.  Re-adding label.

**What was needed:** a lengthy timeout in the section of the test that waits until at least one pod is running and ready.  60 seconds is too short and guarantees flakiness.

**Comment by janetkuo** [comment]
Need to cherrypick this, see https://github.com/kubernetes/kubernetes/pull/33209#issuecomment-248753341

**Comment by jessfraz** [comment]
why does a "refactor" need to be cherry-pick, I would have preferred a bug fix separate from a refactor

**Comment by davidopp** [comment]
#33454 describes the full set of PRs involved, but yeah, maybe we don't actually need to do cherrypicks. Let me think about this. Will follow up on #33454.

**Comment by jessfraz** [comment]
I'm going to remove the cherry-pick candidate label since the changes were in #33251

**Pull request merged**